Sources/AWSLambdaRuntimeCore/LambdaHandler.swift

```diff
@@ -164,6 +164,19 @@ public protocol ByteBufferLambdaHandler {
     /// - Returns: An `EventLoopFuture` to report the result of the Lambda back to the runtime engine.
     ///            The `EventLoopFuture` should be completed with either a response encoded as `ByteBuffer` or an `Error`
     func handle(context: Lambda.Context, event: ByteBuffer) -> EventLoopFuture<ByteBuffer?>
+
+    /// The method to clean up your resources.
+    /// Concrete Lambda handlers implement this method to shutdown their `HTTPClient`s and database connections.
+    ///
+    /// - Note: In case your Lambda fails while creating your LambdaHandler in the `HandlerFactory`, this method
+    ///         **is not invoked**. In this case you must cleanup the created resources immediately in the `HandlerFactory`.
+    func shutdown(context: Lambda.ShutdownContext) -> EventLoopFuture<Void>
```

Comment:
Is this invoked once per lambda invocation?

Reply:
no. only in debug mode for shutdown. or if the lambda occurs an recoverable error. and the lambda is therefore closed.

Reply:
@fabianfett ah right. In that case, couldn't we run it synchronously _after_ `MTELG.withCallingThreadAsEventLoop` returns?

Reply:
Ah no, doesn't make sense. Shutdown probably needs the EL :)

Reply:
ignore my comments. This looks good.